lib/BitcastUtils.cpp

```diff
@@ -898,31 +898,24 @@ bool IsImplicitCasts(Module &M, DenseMap<Value *, Type *> &type_cache,
     source_ty = clspv::InferType(source, M.getContext(), &type_cache);
     dest_ty = st->getValueOperand()->getType();
   } else if (auto *phi = dyn_cast<PHINode>(&I)) {
-    if (phi->getNumIncomingValues() > 0) {
-      auto RefOp = phi->getIncomingValue(0);
-      auto RefOpTy = clspv::InferType(RefOp, M.getContext(), &type_cache);
-      for (unsigned i = 1; i < phi->getNumIncomingValues(); i++) {
+    auto phiImplicitCast = [&M, &type_cache, &phi, &source, &source_ty,
```

Comment:
Feels strange to define a lambda only for here. Am I missing a reason to just have this be functionality?

Reply:
I used a lambda because it felt easier to manage to control flow with return.
Going back to an inlined implementation